### ISSUE TITLE
Failed to run Operator with: operator-sdk run --olm --operator-version 0.1.0 --install-mode AllNamespaces=""

### ISSUE THREAD

**Issue by CohenAviad** [operator-framework/operator-sdk#3001]
## Bug Report

**What did you do?**
My operator watches all namespaces (WATCH_NAMESPACE="").
I run my operator using the following command:
`operator-sdk run --olm --operator-version 0.1.0 --install-mode AllNamespaces=""`

And it got stuck on:
```
INFO[0000] loading Bundles                               dir=deploy/olm-catalog/zoperator
INFO[0000] directory                                     dir=deploy/olm-catalog/zoperator file=zoperator load=bundles
INFO[0000] directory                                     dir=deploy/olm-catalog/zoperator file=0.1.0 load=bundles
INFO[0000] found csv, loading bundle                     dir=deploy/olm-catalog/zoperator file=zoperator.v0.1.0.clusterserviceversion.yaml load=bundles
INFO[0000] loading bundle file                           dir=deploy/olm-catalog/zoperator/0.1.0 file=zadara.com_applicationclones_crd.yaml load=bundle
INFO[0000] loading bundle file                           dir=deploy/olm-catalog/zoperator/0.1.0 file=zadara.com_applicationsnapshots_crd.yaml load=bundle
INFO[0000] loading bundle file                           dir=deploy/olm-catalog/zoperator/0.1.0 file=zadara.com_cloneconfigurations_crd.yaml load=bundle
INFO[0000] loading bundle file                           dir=deploy/olm-catalog/zoperator/0.1.0 file=zadara.com_invokers_crd.yaml load=bundle
INFO[0000] loading bundle file                           dir=deploy/olm-catalog/zoperator/0.1.0 file=zadara.com_snapshotconfigurations_crd.yaml load=bundle
INFO[0000] loading bundle file                           dir=deploy/olm-catalog/zoperator/0.1.0 file=zadara.com_snapshotpolicies_crd.yaml load=bundle
INFO[0000] loading bundle file                           dir=deploy/olm-catalog/zoperator/0.1.0 file=zadara.com_vpsas_crd.yaml load=bundle
INFO[0000] loading bundle file                           dir=deploy/olm-catalog/zoperator/0.1.0 file=zoperator.v0.1.0.clusterserviceversion.yaml load=bundle
INFO[0000] loading Packages and Entries                  dir=deploy/olm-catalog/zoperator
INFO[0000] directory                                     dir=deploy/olm-catalog/zoperator file=zoperator load=package
INFO[0000] directory                                     dir=deploy/olm-catalog/zoperator file=0.1.0 load=package
INFO[0000] Registry data is current                     
INFO[0000] Creating resources                           
INFO[0000]   Creating CatalogSource "default/zoperator-ocs" 
INFO[0000]   Creating Subscription "default/zoperator-v0-1-0-sub" 
INFO[0000]   Creating OperatorGroup "default/operator-sdk-og" 
INFO[0000] Waiting for ClusterServiceVersion "default/zoperator.v0.1.0" to reach 'Succeeded' phase 
INFO[0000]   Waiting for ClusterServiceVersion "default/zoperator.v0.1.0" to appear 
```

**What did you expect to see?**
Generally, my operator finally running. 
The reason my operator didn't run is because the csv was not installed due to wrong OperatorGroup spec:
 ```
$ kubectl describe og
Name:         operator-sdk-og
Namespace:    default
Labels:       <none>
Annotations:  <none>
API Version:  operators.coreos.com/v1
Kind:         OperatorGroup
Metadata:
  Creation Timestamp:  2020-05-11T05:41:40Z
  Generation:          1
  Resource Version:    13817250
  Self Link:           /apis/operators.coreos.com/v1/namespaces/default/operatorgroups/operator-sdk-og
  UID:                 13a7818f-fb74-422a-85c0-e457ae2b1041
Spec:
  Target Namespaces:
    
Events:  <none>
```

As mentioned in the [docs](https://docs.openshift.com/container-platform/4.2/operators/understanding_olm/olm-understanding-operatorgroups.html), when operator watches all namespaces, OperatorGroup spec should be empty:

![image](https://user-images.githubusercontent.com/46089756/81527871-01892800-9364-11ea-9688-aa74b2e292ab.png)

**What did you see instead? Under which circumstances?**
After editing OperatorGroup and left spec empty, csv was installed and operator ran and watches all namespaces as expected.

```
$ kubectl describe og
Name:         operator-sdk-og
Namespace:    default
Labels:       <none>
Annotations:  olm.providedAPIs:
                ApplicationClone.v1alpha1.zadara.com,ApplicationSnapshot.v1alpha1.zadara.com,CloneConfiguration.v1alpha1.zadara.com,Invoker.v1alpha1.zadar...
API Version:  operators.coreos.com/v1
Kind:         OperatorGroup
Metadata:
  Creation Timestamp:  2020-05-11T05:41:40Z
  Generation:          3
  Resource Version:    13818251
  Self Link:           /apis/operators.coreos.com/v1/namespaces/default/operatorgroups/operator-sdk-og
  UID:                 13a7818f-fb74-422a-85c0-e457ae2b1041
Spec:
Status:
  Last Updated:  2020-05-11T05:49:12Z
  Namespaces:
    
Events:  <none>
```

**Environment**
* operator-sdk version:

```
$ operator-sdk version
operator-sdk version: "v0.16.0", commit: "55f1446c5f472e7d8e308dcdf36d0d7fc44fc4fd", go version: "go1.13.8 linux/amd64"
```

* go version:

```
$ go version
go version go1.13.7 linux/amd64
```

* Kubernetes version information:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:14:22Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:07:13Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
* Kubernetes cluster kind: 
```

* Kubernetes cluster kind: 

Operator is written in go

**Possible Solution**
The operator-sdk should generate the OperatorGroup with empty spec if --install-mode AllNamespaces="" is given.

**Comment by estroz** [comment]
This is a bug that was fixed in v0.17.0. If this is back-portable I will create a v0.16.1 release.

/kind bug
/assign

**Comment by estroz** [comment]
@CohenAviad this bug was fixed in v0.17.1. Can you use that version or do you need a patch release of v0.16?

**Comment by CohenAviad** [comment]
@estroz 
thanks for the quick response. 

I verified the bug isn't reproduced in operator-sdk v0.17.1. I will use that version, no reason for a patch.
